src/Shared/LayoutRenderers/AspNetRequestUrlRenderer.cs

```diff
@@ -19,42 +18,123 @@ namespace NLog.Web.LayoutRenderers
     /// </summary>
     /// <remarks>
     /// <code>
-    /// ${aspnet-request-url:IncludeQueryString=true} - produces http://www.exmaple.com/?t=1
-    /// ${aspnet-request-url:IncludeQueryString=false} - produces http://www.exmaple.com/
-    /// ${aspnet-request-url:IncludePort=true} - produces http://www.exmaple.com:80/
-    /// ${aspnet-request-url:IncludePort=false} - produces http://www.exmaple.com/
-    /// ${aspnet-request-url:IncludeScheme=false} - produces www.exmaple.com/
-    /// ${aspnet-request-url:IncludePort=true:IncludeQueryString=true} - produces http://www.exmaple.com:80/?t=1
+    /// ${aspnet-request-url:IncludeQueryString=true} - produces http://www.example.com/?t=1
+    /// ${aspnet-request-url:IncludeQueryString=false} - produces http://www.example.com/
+    /// ${aspnet-request-url:IncludePort=true} - produces http://www.example.com:80/
+    /// ${aspnet-request-url:IncludePort=false} - produces http://www.example.com/
+    /// ${aspnet-request-url:IncludeScheme=false} - produces www.example.com/
+    /// ${aspnet-request-url:IncludePort=true:IncludeQueryString=true} - produces http://www.example.com:80/?t=1
     /// </code>
     /// </remarks>
     /// <seealso href="https://github.com/NLog/NLog/wiki/AspNetRequest-Url-Layout-Renderer">Documentation on NLog Wiki</seealso>
     [LayoutRenderer("aspnet-request-url")]
     public class AspNetRequestUrlRenderer : AspNetLayoutRendererBase
     {
+        /// <summary>
+        /// A flags enumeration that controls which of the five portions of the URL are logged.
+        /// Defaults to scheme://host/path, port and query string are by default not logged.
+        /// </summary>
+        public AspNetRequestUrlProperty Properties { get; set; } = AspNetRequestUrlProperty.Default;
+
         /// <summary>
         /// To specify whether to include / exclude the Query string. Default is false.
         /// </summary>
-        public bool IncludeQueryString { get; set; }
+        [Obsolete("Please use the Properties flags enumeration instead")]
+        public bool IncludeQueryString 
+        {
+            get => (Properties & AspNetRequestUrlProperty.QueryString) == AspNetRequestUrlProperty.QueryString;
+            set
+            {
+                if (value)
```

Comment:
I know this is obsolete code, but could the setter call a private method, that takes two parameters as input (boolean + property-enum). Then the same private method could be used for all the obsolete-boolean-settings.

Reply:
I see yes

Reply:
also did a helper for the getter, that reduced the duplicated code also.